drizzlepac/hlautils/astroquery_utils.py

```diff
@@ -44,65 +46,77 @@ def retrieve_observation(obsid, suffix=['FLC'], archive=False,clobber=False):
+    # filtering again.
+    if data_products_by_id:
+        log.info(
+            "WARNING: No FLC files found for {} - will look for FLT files "
+            "instead.".format(obsid))
         suffix = ['FLT']
-        dataProductsByID = Observations.filter_products(dpobs,
-                                              productSubGroupDescription=suffix,
-                                              extension='fits',
-                                              mrp_only=False)
-
-        # If still no data, then return.  An exception will eventually be thrown in
-        # the higher level code.
-        if len(dataProductsByID) == 0:
-            log.info("WARNING: No FLC or FLT files found for {}.".format(obsid))
+        data_products_by_id = \
+            Observations.filter_products(dpobs,
+                                         productSubGroupDescription=suffix,
+                                         extension='fits',
+                                         mrp_only=False)
+
+        # If still no data, then return.  An exception will eventually be
+        # thrown in the higher level code.
+        if data_products_by_id:
+            log.info(
+                "WARNING: No FLC or FLT files found for {}.".format(obsid))
             return local_files
-    allImages = []
-    for tableLine in dataProductsByID:
-        allImages.append(tableLine['productFilename'])
-    log.info(allImages)
+    all_images = []
+    for tableLine in data_products_by_id:
+        all_images.append(tableLine['productFilename'])
+    log.info(all_images)
     if not clobber:
-        rowsToRemove = []
-        for rowCtr in range(0,len(dataProductsByID)):
-            if os.path.exists(dataProductsByID[rowCtr]['productFilename']):
-                log.info("{} already exists. File download skipped.".format(dataProductsByID[rowCtr]['productFilename']))
-                rowsToRemove.append(rowCtr)
-        if rowsToRemove:
-            rowsToRemove.reverse()
-            for rowNum in rowsToRemove:
-                dataProductsByID.remove_row(rowNum)
-
-    manifest = Observations.download_products(dataProductsByID, mrp_only=False)
+        rows_to_remove = []
+        for rowCtr in range(0, len(data_products_by_id)):
+            if os.path.exists(data_products_by_id[rowCtr]['productFilename']):
+                log.info("{} already exists. File download skipped.".format(
+                    data_products_by_id[rowCtr]['productFilename']))
+                rows_to_remove.append(rowCtr)
+        if rows_to_remove:
+            rows_to_remove.reverse()
+            for rowNum in rows_to_remove:
+                data_products_by_id.remove_row(rowNum)
+
+    manifest = Observations.download_products(data_products_by_id, 
+                                              mrp_only=False)
 
     if not clobber:
-        rowsToRemove.reverse()
-        for rownum in rowsToRemove:
+        rows_to_remove.reverse()
+        for rownum in rows_to_remove:
```

Comment:
Here the above two statements can be replaced with
```python
for rownum in rows_to_remove[::-1]:
```

Reply:
Issue resolved in commit [63eaeb1](https://github.com/spacetelescope/drizzlepac/pull/307/commits/63eaeb1e2686d5afa03e471ef5e53d3c997502fa)